TrigScint/src/TrigScint/TrigScintDigiProducer.cxx

```diff
@@ -137,9 +130,11 @@ void TrigScintDigiProducer::produce(framework::Event &event) {
     Ypos[id] = Ypos[id] / Edep[id];
     Zpos[id] = Zpos[id] / Edep[id];
     double meanPE = depEnergy / mevPerMip_ * pePerMip_;
-    cellPEs[id] = random_->Poisson(meanPE + meanNoise_);
+    std::poisson_distribution<int> poisson_dist(meanPE + meanNoise_);
+    cellPEs[id] = poisson_dist(rng_);
 
     // If a cell has a PE count above threshold, persit the hit.
+    // TAV: we should move the threshold to be an input variable
```

Comment:
I think this is 1 PE as it is the minimum thinkable quantum so I'm not so sure it needs to be configurable

Reply:
no. we let the hits be whatever they are. thresholds are introduced (and configurable) in clustering. 

as a bit of context, only the electronics noise (SiPM shot noise) is this low, and, indeed it is this low. this cut only suppresses artifical noise that is below one light quantum in the SiPM and unphysical.

Reply:
Thanks this is great, I'll add this comment in the code